pkg/dataset/manager.go

```diff
@@ -1,51 +1,55 @@
 package dataset
 
 import (
+	"context"
 	"crypto/rand"
 	"encoding/json"
 	"fmt"
-	"os"
-	"path/filepath"
+	"strings"
 
-	"github.com/gptscript-ai/datasets/pkg/util"
+	"github.com/gptscript-ai/go-gptscript"
 )
 
 const (
-	datasets  = "datasets"
-	extension = ".dataset.json"
+	prefix            = "datasets/"
+	datasetFilePrefix = prefix + "dataset-"
 )
 
 type Manager struct {
-	datasetDir string
+	gptscriptClient *gptscript.GPTScript
+	workspaceID     string
 }
 
-func NewManager(workspaceDir string) (Manager, error) {
-	datasetDir := filepath.Join(workspaceDir, datasets)
-	if _, err := os.Stat(datasetDir); os.IsNotExist(err) {
-		if err := os.MkdirAll(datasetDir, 0755); err != nil {
-			return Manager{}, fmt.Errorf("failed to create dataset directory: %w", err)
-		}
+func NewManager() (Manager, error) {
+	g, err := gptscript.NewGPTScript()
+	if err != nil {
+		return Manager{}, fmt.Errorf("failed to create GPTScript: %w", err)
 	}
 
-	return Manager{datasetDir: datasetDir}, nil
+	return Manager{gptscriptClient: g}, nil
 }
 
-func (m *Manager) ListDatasets() ([]DatasetMeta, error) {
-	files, err := filepath.Glob(filepath.Join(m.datasetDir, "*"+extension))
+func (m *Manager) ListDatasets(ctx context.Context) ([]DatasetMeta, error) {
+	files, err := m.gptscriptClient.ListFilesInWorkspace(ctx, gptscript.ListFilesInWorkspaceOptions{
+		Prefix:      datasetFilePrefix,
```

Comment:
I would be surprised if this worked in general. The intention with the `Prefix` fields (here and others) is that they are "directories." If we need more general prefixes, then I believe we would have to add support for it.

Reply:
At least with the directory provider, this works. I have not tested it with S3 yet.

Reply:
Interesting. It is the directory provider that I figured would not work.